spring-integration-amqp/src/main/java/org/springframework/integration/amqp/inbound/AmqpInboundChannelAdapter.java

```diff
@@ -290,7 +290,7 @@ private void setupRecoveryCallbackIfAny() {
 						context -> {
 							Message messageToRecover =
 									(Message) context.getAttribute(AmqpMessageHeaderErrorMessageStrategy.AMQP_RAW_MESSAGE);
-							if (messageToRecover != null) {
+							if (messageToRecover != null && context.getLastThrowable() != null) {
```

Comment:
Please, hold off these changes.
I'm aware of the failure over there after Spring AMQP migration to Spring Core Retry.
Stay tuned: will PR soon enough.
Thanks